runtime/src/main/java/org/corfudb/runtime/FastSmrMapsLoader.java

```diff
@@ -0,0 +1,319 @@
+ */
+
+@Slf4j
+@Accessors(chain = true)
+public class FastSmrMapsLoader {
+
+    static final long DEFAULT_BATCH_FOR_FAST_LOADER = 5;
+    static final int DEFAULT_TIMEOUT_MINUTES_FAST_LOADING = 30;
+
+    private CorfuRuntime runtime;
+
+    @Setter
+    @Getter
+    private boolean loadInCache;
+
+    @Setter
+    @Getter
+    private long logHead = -1;
+
+    @Setter
+    @Getter
+    private long logTail = -1;
+
+    @Setter
+    @Getter
+    private long batchReadSize = DEFAULT_BATCH_FOR_FAST_LOADER;
+
+    @Setter
+    @Getter
+    private int timeoutInMinutesForLoading = DEFAULT_TIMEOUT_MINUTES_FAST_LOADING;
+
+    // In charge of summoning Corfu maps back in this world
+    private ExecutorService necromancer;
+
+    private Map<UUID, StreamMetaData> streamsMetaData;
+
+    public FastSmrMapsLoader(@Nonnull final CorfuRuntime corfuRuntime) {
+        this.runtime = corfuRuntime;
+        loadInCache = !corfuRuntime.cacheDisabled;
+        streamsMetaData = new HashMap<>();
+        necromancer = Executors.newSingleThreadExecutor();
+    }
+
+    /** Check if this entry is relevant
+     *
+     * Entries before checkpoint starts are irrelevant since they are
+     * contained in the checkpoint.
+     *
+     * @param streamId identifies the Corfu stream
+     * @param entry entry to potentially apply
+     * @return if we need to apply the entry.
+     */
+    private boolean shouldEntryBeApplied(UUID streamId, SMREntry entry) {
+        if (!streamsMetaData.containsKey(streamId)) {
+            return true;
+        }
+
+        return (entry.getEntry().getGlobalAddress() >=
+                streamsMetaData.get(streamId).getHeadAddress());
+    }
+
+    /** Update the corfu object and it's underlying stream with the new entry.
+     *
+     * @param streamId identifies the Corfu stream
+     * @param entry entry to apply
+     */
+    private void applySmrEntryToStream(UUID streamId, SMREntry entry) {
+        if (shouldEntryBeApplied(streamId, entry)) {
+            createSmrMapIfNotExist(streamId);
+
+            ObjectsView.ObjectID thisObjectId = new ObjectsView.ObjectID(streamId, SMRMap.class);
+            CorfuCompileProxy cp = ((CorfuCompileProxy) ((ICorfuSMR) runtime.getObjectsView().getObjectCache().get(thisObjectId)).
+                    getCorfuSMRProxy());
+
+            cp.getUnderlyingObject().applyUpdateToStreamUnsafe(entry);
+        }
+    }
+
+    /** Print the anatomy of a LogData
+     *
+     * Print how many streams are contained in the Metadata and
+     * how many entries per stream.
+     *
+     * Pretty useful for understanding how the the db is being used
+     * from the application perspective.
+     *
+     * @param logData
+     */
+    private void printLogAnatomy(ILogData logData) {
+        try {
+            LogEntry le = deserializeLogData(logData);
+            if (le.getType() == LogEntry.LogEntryType.SMR) {
+                System.out.println("Number of Streams: 1");
```

Comment:
This function is just a utility function. It is not called during the normal process. I left it there for printing the Log Anatomy of a log if somebody want to use it in development.